### PR TITLE
Created selector widgets for state and county for querying SBU database

### PR DESCRIPTION
**Pull request by owenchen04**
## Overview

Added selectors for state and county to retrieve relevant data from SBU database

## Replication

Run requirements.txt, run the streamlit app and test selectors directly on the website.

## Changes Made

Reused code from Using the SBU Database.pynb and implemented selector functionality. Updated map to work with streamlit.
Eliminated hardcoded Tompkins, NY data and replaced with user input. 

## Test Coverage

We tried out different states and counties.

## Screenshots

<details>

  <summary>Full screen of website with selectors and data</summary>

Screenshot 2023-10-22 161653

</details>

### CODE REVIEW

**Review by hehub000 — COMMENTED** [review]
Good job! Here are my comments
- Try to be more consistent with variable naming. For example, you have a function named format_county, and then you redefine it as a variable. You also have a dataframe named df. Things like these can make it unclear what variables represent
- You mentioned that you tested different states and counties, but is it possible that there could be some user input that leads to a query that returns something incorrect? If this might be a possibility, consider making a helper function that can test all states and counties to make sure this code doesn't require error handling

File: get_sbu_data.py [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    'Select a county',
+    counties)
+
+st.write('You selected:', selected_county, ", ", selected_state)
+
+def format_county(selected_county):
+    '''
+    Returns the formatted county string from uppercase to lowercase and capitalized
+    Example: format_county('SAN FRANCISCO') returns 'San Francisco'
+    '''
```
Comment (LiuBrianna) [inline get_sbu_data.py:39]: My suggestion here is to have the function spec be more insightful and keeps the parameter name consistent. The spec is a bit confusing without the example. Maybe reword it to be "Returns the title capitalization of the selected_county".

File: get_sbu_data.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+For instance, sometimes county names are listed as just "Tompkins"; sometimes they are listed as "Tompkins County".
+A more robust version utilizes another approach. See below:
+"""
+# Get county names based on Steve's work. If we know our county of interest is Tompkins, NY, we can search for other names it may be listed under
+counties = pd.read_csv("https://raw.githubusercontent.com/edgi-govdata-archiving/ECHO_modules/main/data/state_counties_corrected.csv") # Get a county name lookup
+counties = counties.groupby(by=["FAC_STATE", "County", "FAC_COUNTY"]).count() # Restructure the data
+counties = counties.iloc[(counties.index.get_level_values('County') == selected_county) & (counties.index.get_level_values('FAC_STATE') == selected_state)].reset_index() # Search for COUNTY, STATE
+counties = list(counties["FAC_COUNTY"].unique()) # List FAC_COUNTY names in ECHO associated with COUNTY
+# Format a sql query
+c = "("
```
Comment (LiuBrianna) [inline get_sbu_data.py:62]: This variable name is not very insightful to describe the functionality of it.

File: get_sbu_data.py [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+for county in counties:
+  c += "\'"+county+"\',"
+c = c[:-1] + ")"
+sql = 'select * from "ECHO_EXPORTER" where "FAC_COUNTY" in ' + c + ' and "FAC_STATE" = \'' + selected_state + '\'' # "ECHO_EXPORTER" contains basic info about all regulated industrial facilities, including location
+print(sql)
```
Comment (LiuBrianna) [inline get_sbu_data.py]: Please correct me if I am wrong, but do you need this print statement? Was this for debugging?

File: get_sbu_data.py [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+            text = row["FAC_NAME"] + ' - '
+        except TypeError:
+            print( "A facility was found without a name. ")
+        if 'DFR_URL' in row:
+            text += " - <p><a href='"+row["DFR_URL"]
+            text += "' target='_blank'>Link to ECHO detailed report</a></p>" 
+    return text
+
+def bivariate_map(regions, points, bounds=None, no_text=False):
+    '''
```
Comment (LiuBrianna) [inline get_sbu_data.py:113]: There is an inconsistent formatting for the function specs. Try to keep them organized the same way in order for other coders to understand what your function does.

File: get_sbu_data.py [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+    show the map of region(s) (e.g. zip codes) and points (e.g. facilities within the regions)
+    create the map using a library called Folium (https://github.com/python-visualization/folium)
+    bounds can be preset if necessary
+    no_text errors can be managed
+    '''
+    m = folium.Map()  
+
```
Comment (LiuBrianna) [inline get_sbu_data.py:120]: To reiterate, there one letter variable names are a bit confusing and lack insight for their intended purpose.

**Review by LiuBrianna — COMMENTED** [review]
Overall, the code looks good. There are some minor code things to ensure good code quality.

**Comment by jessicajcho** [comment]
Overall good job! In the future, make sure to resolve the code review comments manually (by pressing the resolve conversation) & remove unneeded code
